fs/cromfs/fs_cromfs.c

```diff
@@ -988,7 +988,7 @@ static int cromfs_opendir(FAR struct inode *mountpt, FAR const char *relpath,
 
   /* Verify that the node is a directory */
 
-  if (!S_ISDIR(node->cn_mode))
+  if (!S_ISDIR(node->cn_mode) && !S_ISLNK(node->cn_mode))
```

Comment:
for link we need look at the target to decide whether is a directory, If we don't want to support the link now, it's better to:
if (S_ISLNK(node->cn_mode))
  {
    /* Not implement yet */

    return -ENOSYS;
  }
else if (!S_ISDIR(node->cn_mode))
  {
    return -ENOTDIR;
  }

Reply:
@xiaoxiang781216 

Results of your suggested change are below. It breaks it again.

Please merge this PR. It is correct in the context of the cromfs and gencromfs.c 

```
nsh> ls -R /
/:
 dev/
 etc/
 fs/
 obj/
 proc/
/dev:
 baro0
 console
 console_buf
 led0
 mag0
 mag1
 mmcsd0
 mtdblock0
 mtdblock1
 null
 tone_alarm0
 ttyACM0
 ttyS0
 ttyS1
 ttyS2
 ttyS3
 ttyS4
 ttyS5
 ttyS6
 uavcan/
/dev/uavcan:
 airspeed
 baro
 battery
 differential_pressure
 esc
 flow
 gnss
 mag
/etc:
nsh: ls: no such directory: /etc
nsh>

```
I think you need to understand the the ripple effects the change you made to the total set of fs's. I have tried to explain it it in the other PR and it was clear for your response, we are not communicating clearly.  

I understand that following the link is the {uni|pos}ix way, but the PR 1372 `vfs/stat: Make the flag defintion more confirm POSIX standard.` broke the nuttx way on **_ALL_** fs's that depended on combinatorial operations. [see Greg's comment](https://github.com/apache/incubator-nuttx/pull/1522#issuecomment-670223690)


Please test master on the sim with a cromfs and all the others fs that may have depended on the following operations

`(m) & TYPE_MASK == TYPE` and `(m) & LINK_MASK == LINK` were m is (TYPE|LINK)

Reply:
I already check the whole base to ensure there isn't | with S_IFLNK before send the PR. The problem is that genromfs use the different name to identify the link which is hard to search.

Reply:
BTW, if you are using CROMFS, please add a config to enable the automation build. If you don't take care, nobody can ensure it will always build and work as expect. As I suggest you before to enable the out of board test.

Reply:
@xiaoxiang781216 - Yes I know, but the important point is **there may be more of them**!

Reply:
After comparing gencromfs.c and https://github.com/chexum/genromfs/blob/master/genromfs.c, it's better to change the link to normal directory:
https://github.com/apache/incubator-nuttx/blob/master/tools/gencromfs.c#L1019
https://github.com/apache/incubator-nuttx/blob/master/tools/gencromfs.c#L1020
https://github.com/apache/incubator-nuttx/blob/master/tools/gencromfs.c#L1375
because ROMFS is done like this:
https://github.com/chexum/genromfs/blob/master/genromfs.c#L810
https://github.com/chexum/genromfs/blob/master/genromfs.c#L823

Reply:
@xiaoxiang781216 Why would @patacongo done the extra work, if it was not needed? The operation on the cromfs look very different to me (one is recursive, the other is not). If you do do a separate PR, I would be happy to test it.

Reply:
I am fine with the current change, @patacongo please merge it if you don't have more concern.

Reply:
> BTW, if you are using CROMFS, please add a config to enable the automation build. If you don't take care, nobody can ensure it will always build and work as expect. As I suggest you before to enable the out of board test.

Done but I need some info.... 
https://github.com/apache/incubator-nuttx/pull/1532

Reply:
> As I suggest you before to enable the out of board test.

I hope you meant on the sim.